### PR TITLE
Custom template

### PR DESCRIPTION
**Pull request by ludovico85**
How to customize the default lizmap theme by modifying the modules templates.

* Ticket : #2532
*

### CODE REVIEW

File: source/publish/customization/themes.rst [hunk 1]
```diff
@@ -110,3 +110,50 @@ for :file:`css/main.css` and:
 for :file:`css/map.css`
 
 By following these steps, we keep our custom theme as small as possible.
+
+======================
+Custom templates
+======================
+
+In Lizmap each module (i.e., view, admin, action, etc.) defines some template. Templates of each module are located in :file:`lizmap/modules` folder. Each module has one or more templates (.tpl) in the templates folder :file:`lizmap/modules/moduleX/templates/`. In order to customize the template of a module it is advisable to create a copy of the files without modify the original files. To do this, it is necessary to create a copy of the templates inside the default theme's folder of Lizmap, which is in :file:`var/themes/default`.
```
Comment (nboisteault) [inline source/publish/customization/themes.rst:118]: some template => some templates

File: source/publish/customization/themes.rst [hunk 2]
```diff
@@ -110,3 +110,50 @@ for :file:`css/main.css` and:
...
+
+Prerequisites
+-------------
+
+* A folder with the same name of the module to which the template to redifine belongs, in the :file:`var/themes/default` folder.
+
+Configuring the tool
+--------------------
+Simply copy the template to redifine in the module folder and custmize it with a text editor.
```
Comment (nboisteault) [inline source/publish/customization/themes.rst:127]: redifine => redefine
custmize => customize

File: source/publish/customization/themes.rst [hunk 3]
```diff
@@ -110,3 +110,50 @@ for :file:`css/main.css` and:
...
+
+Example
+-------
+We want to change the default title in the header of the main page of lizmap. The "view" module and the :file:`main.tpl` template are involved in this procedure. We can see the name of the module in the URL bar (``myhost/lizmap/index.php/view/``).
+
+* Create a directory named view in the theme's default folder
+
+.. code-block:: none
+nano mkdir lizmap/var/themes/default/view
```
Comment (nboisteault) [inline source/publish/customization/themes.rst:136]:
```
.. code-block:: bash
  nano mkdir lizmap/var/themes/default/view
```
two spaces before `nano`

File: source/publish/customization/themes.rst [hunk 4]
```diff
@@ -110,3 +110,50 @@ for :file:`css/main.css` and:
...
+
+* Copy the :file:`main.tpl` file from the default location in :file:`lizmap/modules/view/templates/` to the :file:`lizmap/var/themes/default/view` folder
+
+.. code-block:: none
+cp lizmap/var/themes/default/view/main.tpl lizmap/var/themes/default/view
```
Comment (nboisteault) [inline source/publish/customization/themes.rst:141]:
```
.. code-block:: bash
  cp lizmap/modules/view/main.tpl lizmap/var/themes/default/view
```
two spaces before `cp`

File: source/publish/customization/themes.rst [hunk 5]
```diff
@@ -110,3 +110,50 @@ for :file:`css/main.css` and:
 for :file:`css/map.css`
 
 By following these steps, we keep our custom theme as small as possible.
+
+======================
+Custom templates
+======================
```
Comment (nboisteault) [inline source/publish/customization/themes.rst:116]:
```
Custom templates
================
```

File: source/publish/customization/themes.rst [hunk 6]
```diff
@@ -110,3 +110,50 @@ for :file:`css/main.css` and:
...
+
+In Lizmap each module (i.e., view, admin, action, etc.) defines some template. Templates of each module are located in :file:`lizmap/modules` folder. Each module has one or more templates (.tpl) in the templates folder :file:`lizmap/modules/moduleX/templates/`. In order to customize the template of a module it is advisable to create a copy of the files without modify the original files. To do this, it is necessary to create a copy of the templates inside the default theme's folder of Lizmap, which is in :file:`var/themes/default`.
+
+Prerequisites
+-------------
+
+* A folder with the same name of the module to which the template to redifine belongs, in the :file:`var/themes/default` folder.
```
Comment (nboisteault) [inline source/publish/customization/themes.rst:123]: redifine => redefine

**Review by nboisteault — CHANGES_REQUESTED** [review]
Thank you very much @ludovico85 for your PR! It will help a lot Lizmap users.
Can you make those little changes?
I use Code Spell Checker for Visual studio code to find grammar issues.

**Review by Gustry — APPROVED** [review]
Thanks !

**Review by nboisteault — CHANGES_REQUESTED** [review]
`.. code-block:: none` => `.. code-block:: bash` line 135 and 140 and it will be ok.

**Comment by ludovico85** [comment]
> `.. code-block:: none` => `.. code-block:: bash` line 135 and 140 and it will be ok.

Solved

**Pull request merged**